docs/retail_versions.md

```diff
@@ -0,0 +1,166 @@
+files in `tools/disasm/gc-eu-mq` say how to split the source files and where the
+functions and variables are in the target ROM, and these may need to be updated
+if there are mistakes or if function names change due to documentation work.
+
+Unfortunately, the disassembly is not perfect, so a "correct" decompilation might
+still show diffs with data symbols. We might improve this later, but these data
+diffs are fine to ignore for now.
+
+For register and stack allocation differences, often the code can be tweaked so
+that it matches both the retail ROM while continuing to match the Debug ROM (for
+example, by reordering assignments or moving a local variable declaration inside
+an `if` block). Since retail MM versions use the same compiler flags as retail
+OOT, checking MM decomp for similar code can help.
+
+We can disable code that was removed in retail builds by adding
+`#ifdef OOT_DEBUG` around these parts of the code. In order too keep the code
+readable, we should try to minimize the amount of `#ifdef` noise whenever
+possible. For example, instead of writing
+
+```c
+void foo() {
+#ifdef OOT_DEBUG
+    if (condition || SREG(0)) {
+#else
+    if (condition) {
+#endif
+        bar();
+    }
+
+#ifdef OOT_DEBUG
+    if (condition || SREG(0)) {
+#else
+    if (condition) {
+#endif
+        baz();
+    }
+}
+```
+
+consider something like this instead:
+
+```c
+#ifdef OOT_DEBUG
+#define CONDITION (condition || SREG(0))
+#else
+#define CONDITION (condition)
+#endif
+
+void foo() {
+    if (CONDITION) {
+        bar();
+    }
+
+    if (CONDITION) {
+        baz();
+    }
+}
+```
+
+## Setup
+
+1.  Copy your target PAL GameCube Master Quest ROM (non-debug) to
+    `baseroms/gc-eu-mq/baserom.z64`
+
+1.  Extract assets and ROM files from the Debug ROM by running
+
+    ```sh
+    make setup -jN
+    ```
+
+    if necessary.
+
+1.  Build the non-matching test ROM by running
+
+    ```sh
+    make setup -jN VERSION=gc-eu-mq
+    make -jN VERSION=gc-eu-mq
+    ```
+
+    where `N` is the number of cores on your machine. This will build into
+    `build/gc-eu-mq` and produce `oot-gc-eu-mq.z64`.
+
+    If you later want to delete all output files, run
+
+    ```sh
+    make clean VERSION=gc-eu-mq
+    ```
+
+1.  Disassemble the target ROM by running
+
+    ```sh
+    make disasm -jN VERSION=gc-eu-mq
+    ```
+
+    where `N` is the number of cores on your machine. The outputs will be written to
+    `expected/build/gc-eu-mq`.
+
+    Note that if you need to copy a matching build for the Debug ROM, you can use
+
+    ```sh
+    mkdir -p expected/build
+    cp -r build/gc-eu-mq-dbg expected/build
+    ```
+
+    to avoid clobbering the disassembly.
+
+## Diff Tools
+
+Note that many tools will require activating the Python virtual environment
+in your terminal session. To do this, run:
+
+```sh
+source .venv/bin/activate
+```
+
+### retail_progress.py
+
+Running `./retail_progress.py file.c` will attempt to figure out which functions
```

Comment:
not a big deal but I was expecting to just run ``./retail_progress.py graph.c`` and it would magically find the path, but yeah I guess it makes sense if several files have the same name

maybe you should add this to avoid any confusion
```suggestion
Running `./retail_progress.py path/to/file.c` will attempt to figure out which functions
```